io.c

```diff
@@ -696,6 +698,7 @@ static int io_read(struct io_kiocb *req, const struct sqe_submit *s,
 	ret2 = call_read_iter(file, kiocb, &iter);
 	if (!force_nonblock || ret2 != -EAGAIN) {
 		io_rw_done(kiocb, ret2);
+		ret = 0;
```

Comment:
This does not seem needed, as "ret" anything other than zero, would have failed in the path above this call.